uxarray/grid/area.py

```diff
@@ -56,23 +62,29 @@ def calculate_face_area(
     # num triangles is two less than the total number of nodes
     num_triangles = num_nodes - 2
 
+    if coords_type == "spherical":
+        # Preallocate arrays for Cartesian coordinates
+        n_points = len(x)
+        x_cartesian = np.empty(n_points)
+        y_cartesian = np.empty(n_points)
+        z_cartesian = np.empty(n_points)
+
+        # Convert all points to Cartesian coordinates using an explicit loop
+        for i in range(n_points):
+            lon_rad = np.deg2rad(x[i])
+            lat_rad = np.deg2rad(y[i])
+            cartesian = _lonlat_rad_to_xyz(lon_rad, lat_rad)
+            x_cartesian[i], y_cartesian[i], z_cartesian[i] = cartesian
+
+        x, y, z = x_cartesian, y_cartesian, z_cartesian
```

Comment:
I'd suggest passing in `Grid.node_x`, `Grid.node_y` and `Grid.node_z` into this function directly to avoid needing to re-compute it.